api/nodes/permissions.py

```diff
@@ -22,6 +22,50 @@ def has_object_permission(self, request, view, obj):
             return obj.can_edit(auth)
 
 
+class AdminOrPublic(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (Node, User)), 'obj must be a Node or Pointer, got {}'.format(obj)
+        if isinstance(obj, Node):
+            node = obj
+        else:
+            node = Node.load(request.parser_context['kwargs']['node_id'])
+        auth = get_user_auth(request)
+        if request.method in permissions.SAFE_METHODS:
+            return node.is_public or node.can_view(auth)
+        else:
+            return node.has_permission(auth.user, 'admin')
+
+
+class ContributorPermissions(permissions.BasePermission):
```

Comment:
This should be renamed. `Contributor` would probably work best with the existing permissions.